### PR TITLE
Adds Blkom Presence.

### PR DESCRIPTION
**Pull request by GhaithAlHallak8**
### Sorry for the previous pull request but it was messed by mistake, sorry.
image
image
image
image
image

### CODE REVIEW

File: websites/B/Blkom/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "www.animeblkom.net"
+  ],
+  "description": {
+    "en": "Blkom - Arabic website to watch and download Arabic subbed anime and get the latest news about Anime.",
+    "ar_SA": "أنمي بالكوم - موقع عربي متخصص لمشاهدة وتحميل الانمي المترجم باللغة العربية ومعرفة آخر أخبار ومستجدات عالم الأنمي."
+  },
+  "service": "Blkom",
+  "version": "1.0",
+  "logo": "https://i.imgur.com/XW4ouq1.png",
+  "thumbnail": "https://i.imgur.com/UBqcoIN.png",
```
Comment (ririxichan) [inline websites/B/Blkom/dist/metadata.json:19]: This should be screenshot or banner of this website.

File: websites/B/Blkom/presence.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  data.startTimestamp = timestamps[0];
+  data.endTimestamp = timestamps[1];
+  if (video.paused) {
+    delete data.startTimestamp;
+    delete data.endTimestamp;
+  }
+
+  presence.setActivity(data, !video.paused);
+} else if (location.pathname.startsWith('/search')) {
+  data.smallImageKey = "searching";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:55]: you should use searching png from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Ova";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/ona-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Ona";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/special-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:102]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Movie";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/ova-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Ova";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/ona-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:97]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Series";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/movie-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Movie";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/ova-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:92]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/series-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Series";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/movie-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:87]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  data.details = document.querySelector('.name').textContent.trim().slice(0,-5)
+  data.state = "Viewing an Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/anime-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/series-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:82]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  data.state = "Downloading Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/anime/')) {
+  data.smallImageKey = "location";
+  data.smallImageText = "Viewing Anime";
+  data.details = document.querySelector('.name').textContent.trim().slice(0,-5)
+  data.state = "Viewing an Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/anime-list')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:77]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+  else data.state = `Results: Nothing`;
+  presence.setActivity(data);
+} else if (location.pathname.includes('/download')) {
+  data.smallImageKey = "download";
+  data.smallImageText = "Downloading";
+  data.details = document.querySelector('.heading > a').textContent
+  data.state = "Downloading Anime";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/anime/')) {
+  data.smallImageKey = "location";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:71]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Searching";
+  data.details = `Searching: ${document.querySelector('.heading').textContent.slice(18,-2)}`
+  if (document.querySelectorAll('.content').length && document.querySelectorAll('.page-item').length - 2)
+  data.state = `Results: ${document.querySelectorAll('.content').length * (document.querySelectorAll('.page-item').length - 3)} and More..`;
+  else if (document.querySelectorAll('.content').length)
+  data.state = `Results: ${document.querySelectorAll('.content').length}`;
+  else data.state = `Results: Nothing`;
+  presence.setActivity(data);
+} else if (location.pathname.includes('/download')) {
+  data.smallImageKey = "download";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:65]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Ona";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/special-list')) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Browsing";
+  data.details = "Browsing for Special";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith('/premium')) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:107]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Discovering";
+  data.details = "Discovering Premium";
+  presence.setActivity(data);
+} else if (location.pathname === "/blog") {
+  data.smallImageKey = "blog";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:112]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+  data.details = "Discovering Blog";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/post")) {
+  data.smallImageKey = "blog";
+  data.smallImageText = "Viewing Post";
+  data.details = document.querySelector('.post-title').textContent.trim()
+  data.state = `Viewing ${document.querySelector('.publisher').textContent.trim()}'s Post`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/timeline")) {
+  data.smallImageKey = "discovery";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:123]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Discovering";
+  data.details = "Discovering Premium";
+  presence.setActivity(data);
+} else if (location.pathname === "/blog") {
+  data.smallImageKey = "blog";
+  data.smallImageText = "Discovering";
+  data.details = "Discovering Blog";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/post")) {
+  data.smallImageKey = "blog";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:117]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+  data.details = document.querySelector('.post-title').textContent.trim()
+  data.state = `Viewing ${document.querySelector('.publisher').textContent.trim()}'s Post`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/timeline")) {
+  data.smallImageKey = "discovery";
+  data.smallImageText = "Discovering";
+  data.details = "Discovering Timeline";
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/ratings')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:128]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Viewing Profile";
+  data.details = `Ratings List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/watching')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:134]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/completed')) {
+  data.smallImageKey = "profile";
+  data.smallImageText = "Viewing Profile";
+  data.details = `Completed List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/on-hold')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:146]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/watching')) {
+  data.smallImageKey = "profile";
+  data.smallImageText = "Viewing Profile";
+  data.details = `Watching List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/completed')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:140]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/on-hold')) {
+  data.smallImageKey = "profile";
+  data.smallImageText = "Viewing Profile";
+  data.details = `On-Hold List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/dropped')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:152]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Viewing Profile";
+  data.details = `Dropped List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user") && location.pathname.includes('/plan-to-watch')) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:158]: you should use pngs from .resources folder :)

File: websites/B/Blkom/presence.ts [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+  data.smallImageText = "Viewing Profile";
+  data.details = `Planned List`
+  data.state = `Viewing ${document.querySelector('.profile-usertitle-name').textContent.trim()}'s Profile`;
+  presence.setActivity(data);
+} else if (location.pathname.startsWith("/user")) {
+  data.smallImageKey = "profile";
```
Comment (ririxichan) [inline websites/B/Blkom/presence.ts:164]: you should use pngs from .resources folder :)

File: websites/B/Blkom/dist/metadata.json [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+    "انمي بالكوم",
+    "انمي بلكوم",
+    "انمي",
+    "انيمي",
+    "انيمي بلكوم",
+    "انيمي بالكوم",
+    "Anime",
+    "Anime Blkom",
+    "Blkom Anime",
+    "Blkom"
```
Comment (ririxichan) [inline websites/B/Blkom/dist/metadata.json:33]: no spaces in tags. also tags are "music, videos" etc. etc. not site name.

**Review by ririxichan — CHANGES_REQUESTED** [review]
fix codacy errors and requested changes

File: websites/B/Blkom/dist/metadata.json [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+    "animeblkom.net",
+    "www.blkom.com",
+    "www.animeblkom.net"
+  ],
+  "description": {
+    "en": "Blkom - Arabic website to watch and download Arabic subbed anime and get the latest news about Anime.",
+    "ar_SA": "أنمي بالكوم - موقع عربي متخصص لمشاهدة وتحميل الانمي المترجم باللغة العربية ومعرفة آخر أخبار ومستجدات عالم الأنمي."
+  },
+  "service": "Blkom",
+  "version": "1.0",
```
Comment (ririxichan) [inline websites/B/Blkom/dist/metadata.json:17]: version should be 1.0.0

**Review by ririxichan — CHANGES_REQUESTED** [review]

**Comment by GhaithAlHallak8** [comment]
All done.

**Comment by ririxichan** [comment]
Upload proof of working after deleting small img and good to accept! :)

**Comment by GhaithAlHallak8** [comment]
image
image
image
image
image

**Comment by GhaithAlHallak8** [comment]
But I think you need to fix the download img in the .resources

**Review by ririxichan — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**